dimod/reference/composites/higherordercomposites.py

```diff
@@ -0,0 +1,351 @@
+       sampler.
+
+       >>> linear = {'a': -4.0, 'b': -4.0}
+       >>> quadratic = {('a', 'b'): 3.2, ('a', 'b', 'c'): 1}
+       >>> sampler = dimod.PolyScaleComposite(dimod.HigherOrderComposite(dimod.ExactSolver()))
+       >>> response = sampler.sample_hising(linear, quadratic, scalar=0.5,
+       ...                ignored_terms=[('a','b')])
+
+    """
+
+    def __init__(self, child):
+        if not isinstance(child, PolySampler):
+            raise TypeError("Child sampler must be a PolySampler")
+        self._children = [child]
+
+    @property
+    def children(self):
+        """The child sampler in a list"""
+        return self._children
+
+    @property
+    def parameters(self):
+        param = self.child.parameters.copy()
+        param.update({'scalar': [],
+                      'bias_range': [],
+                      'poly_range': [],
+                      'ignored_terms': [],
+                      })
+        return param
+
+    @property
+    def properties(self):
+        return {'child_properties': self.child.properties.copy()}
+
+    def sample_poly(self, poly, scalar=None, bias_range=1, poly_range=None,
+                    ignored_terms=None, **parameters):
+        """Scale and sample from the given binary polynomial.
+
+        Ff scalar is not given, problem is scaled based on bias and polynomial
+        ranges. See :meth:`.BinaryPolynomial.scale` and
+        :meth:`.BinaryPolynomial.normalize`
+
+        Args:
+            poly (obj:`.BinaryPolynomial`): A binary polynomial.
+
+            scalar (number, optional):
+                Value by which to scale the energy range of the binary polynomial.
+
+            bias_range (number/pair, optional, default=1):
+                Value/range by which to normalize the all the biases, or if
+                `poly_range` is provided, just the linear biases.
+
+            poly_range (number/pair, optional):
+                Value/range by which to normalize the higher order biases.
+
+            ignored_terms (iterable, optional):
+                Biases associated with these terms are not scaled.
+
+            **parameters:
+                Other parameters for the sampling method, specified by
+                the child sampler.
+
+        """
+
+        if ignored_terms is None:
+            ignored_terms = set()
+        else:
+            ignored_terms = {frozenset(term) for term in ignored_terms}
+
+        # scale and normalize happen in-place so we need to make a copy
+        original, poly = poly, poly.copy()
+
+        if scalar is not None:
+            poly.scale(scalar, ignored_terms=ignored_terms)
+        else:
+            poly.normalize(bias_range=bias_range, poly_range=poly_range,
+                           ignored_terms=ignored_terms)
+
+            # we need to know how much we scaled by, which we can do by looking
+            # at the biases
+            try:
+                v = next((v for v, bias in original.items() if bias))
```

Comment:
what if the first one is in ignored terms? - tad worried that tests don't catch this

Reply:
Good catch!

Reply:
It worked because if we detected ignored terms we just recalculate the energies anyway, but I have updated it to calculate the scalar correctly in case that logic changes later.